core/api/src/modules/ops/app.ts

```diff
@@ -1,38 +1,65 @@
-import { api, log } from "actionhero";
+import { api, log, id } from "actionhero";
 import { App, SimpleAppOptions } from "../../models/App";
+import { waitForLock } from "../locks";
 import { OptionHelper } from "../optionHelper";
 
 export namespace AppOps {
   /**
    * Connect to an App
    */
-  export async function connect(app: App, options?: SimpleAppOptions) {
-    const appOptions = await app.getOptions();
+  export async function connect(
+    app: App,
+    options?: SimpleAppOptions,
+    forceReconnect = false
+  ) {
     const { pluginApp } = await app.getPlugin();
-    const connection = api.plugins.persistentConnections[app.guid];
-    if (connection) {
-      await app.disconnect();
-    }
-    if (pluginApp.methods.connect) {
-      log(`connecting to app ${app.name} - ${pluginApp.name} (${app.guid})`);
-      const connection = await pluginApp.methods.connect({
-        app,
-        appGuid: app.guid,
-        appOptions: options ? options : appOptions,
-      });
-      app.setConnection(connection);
-      return connection;
+    if (!pluginApp.methods.connect) return;
+
+    const appOptions = await app.getOptions();
```

Comment:
i'd suggest a `finally` for all lock situations.

Reply:
👍